include/diaspora.php

```diff
@@ -2060,8 +2121,9 @@ private function original_item($guid, $orig_author, $author) {
 
 				if ($r) {
 					// If it is a reshared post from another network then reformat to avoid display problems with two share elements
-					if (self::is_reshare($r[0]["body"], false))
-						$r[0]["body"] = diaspora2bb(bb2diaspora($r[0]["body"]));
+					if (self::is_reshare($r[0]["body"], false)) {
+						$r[0]["body"] = self::replace_people_guid(diaspora2bb(bb2diaspora($r[0]["body"])), $r[0]["author-link"]);
```

Comment:
Same as above: Avoid chaining functions/methods or replace this call with a filter-chain (other alternative).